app/gui2/src/util/callTree.ts

```diff
@@ -210,78 +208,133 @@ export class ArgumentApplication {
+      const arg = interpreted.args[position]
+      assert(!!arg)
+      const pastPositionalArguments = mapOr(lastPositionalArgIndex, true, (i) => position > i)
+
+      if (pastPositionalArguments && arg.argName != null && arg.argName !== nextArgumentName()) {
+        // Named argument that is not in its natural position, and there are no more
+        // positional arguments to emit in the chain. At this point placeholders can be
+        // inserted. We need to figure out which placeholders can be inserted before
+        // emitting this named argument.
+
+        // all remaining arguments must be named, as we are past all positional arguments.
+        const remainingArguments = interpreted.args.slice(position)
+
+        // For each subsequent argument in its current natural position, insert a
+        // placeholder. Do that only if the argument is not defined further in the chain.
+        while ((nextArgument = takeNextArgument())) {
+          const { index, info } = nextArgument
+          const isDefinedFurther = remainingArguments.some((arg) => arg.argName === info.name)
+          if (isDefinedFurther) {
+            putBackArgument(index)
+            break
+          } else {
+            resolvedArgs.push({
+              appTree: arg.appTree.function,
+              argument: ArgumentPlaceholder.WithRetrievedConfig(
+                callId,
+                index,
+                info,
+                ApplicationKind.Prefix,
+                placeholderAlreadyInserted,
+                widgetCfg,
+              ),
+            })
+            placeholderAlreadyInserted = true
+          }
+        }
+
+        // Finally, we want to emit the named argument and remove it from the list of
+        // remaining known params.
+        const { index, info } = takeNamedArgument(arg.argName) ?? {}
+        resolvedArgs.push({
+          appTree: arg.appTree,
           argument: ArgumentAst.WithRetrievedConfig(
-            realArg.argument,
-            argIndex,
-            tryGetIndex(knownArguments, argIndex),
-            kind,
+            arg.argument,
+            index,
+            info ?? unknownArgInfoNamed(arg.argName),
+            ApplicationKind.Prefix,
             widgetCfg,
           ),
         })
       } else {
-        // When name is present, we need to find the argument with the same name in the list of
-        // known arguments.
-        const name = realArg.argName
-        const foundIdx = argumentsLeftToMatch.findIndex((i) => knownArguments?.[i]?.name === name)
-        const argIndex = foundIdx === -1 ? undefined : argumentsLeftToMatch.splice(foundIdx, 1)[0]
-
-        if (argIndex != null && foundIdx === 0)
-          insertPlaceholdersUpto(argIndex, realArg.appTree.function)
-        prefixArgsToDisplay.push({
-          appTree: realArg.appTree,
+        const argument = arg.argName == null ? takeNextArgument() : takeNamedArgument(arg.argName)
+        const { index, info } = argument ?? {}
```

Comment:
Now we're going really wild, because we have `arg` and `argument` - and both are of different kinds :)